File: player/src/scene.cpp
#include <layer.h>
#include <scene.h>

Scene::Scene(SceneBackgroundColor _background, Layer **_layers, uint8_t _layersCount) :
    background(_background), layers(_layers), layersCount(_layersCount) {
}

void Scene::draw() {
  for (uint8_t i = 0; i < layersCount; i++) {
    layers[i]->draw();
  }
}

File: export/lib/model/project/entities/scene.js
const Entity = require('./entity');
const Layers = require('./scene/layers');

class Scene extends Entity {

  constructor(project, json) {
    super(project, json);
    this.name = json.name;
    this.background = json.background;
    this.size = json.size;
  }

  link() {
    super.link();
    this.index = this.project.scenes.models.indexOf(this);
    this.layers = new Layers(this, this.models.filter(model => model.baseType === 'scene/layer'));
    this.layers.link();
  }

}

module.exports = Scene;
